windows-gaming-driver/src/driver/controller.rs

```diff
@@ -223,14 +269,19 @@ impl Controller {
     pub fn io_detach(&mut self) {
         assert!(self.ga != State::Suspending, "trying to exit from a suspending vm?");
         assert!(self.ga != State::Suspended, "trying to exit from a suspended vm?");
-        if !self.io_attached {
-            return;
-        }
-        for i in self.machine_config.usb_devices.iter().enumerate()
-            .filter(|&(_, dev)| !dev.permanent).map(|(i, _)| i) {
-            (&self.monitor).send(QmpCommand::DeviceDel { id: format!("usb{}", i) }).unwrap();
+
+        match self.io_state {
+            IoState::Detached => (),
+            IoState::LightFallback | IoState::LightEntry => self.input.borrow_mut().suspend(),
+            IoState::FullEntry => {
+                for i in self.machine_config.usb_devices.iter().enumerate()
+                    .filter(|&(_, dev)| !dev.permanent).map(|(i, _)| i) {
+                        (&self.monitor).send(QmpCommand::DeviceDel { id: format!("usb{}", i) }).unwrap();
```

Comment:
Weird indentation, would have done it the other way around (indent .filter more).